Btc and canister icp motion proposals (#59)
ic-association/nns-proposals: proposals/governance/20210915T1501Z.md

> # **Enable smart contract canisters to hold ICP**

Reviewer: ```suggestion
# **Enable all principals including canisters to transfer ICP**
```

## **Summary**

> Expanding support for all Canister types to be able to hold and transfer ICP tokens. Currently, for security reasons, only NNS Canisters and users may hold ICP tokens

Reviewer: ```suggestion
Expanding support for all principals, including application canisters, to be able to hold and transfer ICP tokens. Currently only certain NNS canisters and users may transfer ICP tokens
```

> To vote YES on this proposal means that you agree with the [designed proposed here](https://forum.dfinity.org/t/enable-canisters-to-hold-icp/6153/29?u=diegop).

Reviewer: ```suggestion
To vote YES on this proposal means that you agree with the [design proposed here](https://forum.dfinity.org/t/enable-canisters-to-hold-icp/6153/29?u=diegop).
```

If this NNS proposal passes, then there will be subsequent NNS proposals that will follow the design proposed and will have code updates.

If this NNS proposal fails, then this design will not be implemented and the community will go back to the drawing board.

## **Key people involved**

Jens Groth, Bogdan Warinschi, Roman Kashitsyn, Lomesh Dutta

## **Timeline for this community-wide design discussion**

- [x] [Draft plan for feature](https://forum.dfinity.org/t/enable-canisters-to-hold-icp/6153/29?u=diegop) posted on the forum for review: September 10, 2021, 15:00 UTC 
- [x] NNS Proposal submission:  September 15, 2021, 15:00 UTC 
- [ ] NNS Proposal expiration:  September 17, 2021, 15:00 UTC 

## **Design doc for the community: Enable all principals to transfer ICP**

You can also view doc here: [Proposed design to review and vote on.](https://forum.dfinity.org/t/enable-canisters-to-hold-icp/6153/29?u=diegop)

NNS Proposal: Enable smart contract canisters to hold ICP

> Main authors: Jens Groth

Reviewer: ```suggestion
```
